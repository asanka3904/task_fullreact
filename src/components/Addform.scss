.form-contain {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;

  .add-form {
    margin-bottom: 40px;

    .form-control {
      margin: 20px 0;
    }

    label {
      display: block;
    }

    input {
      width: 100%;
      height: 35px;
      margin: 5px;
      padding: 3px 7px;
      font-size: 17px;
      background: linear-gradient(
        to right bottom,
        rgba(255, 255, 255, 0.7),
        rgba(255, 255, 255, 0.3)
      );
    }

    #button {
      background-color: rgb(65, 41, 172);

      border-radius: 40px;
    }

    label {
      flex: 1;
    }

    input {
      flex: 2;
    }
  }
}
