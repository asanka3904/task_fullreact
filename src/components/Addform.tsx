import * as React from "react";
import { useState } from "react";
import { SaveTodo, Todo } from "../interface/Todo";
import { useAppSelector, useAppDispatch } from "../app/hooks";
import { addTodo, removeTodo, setTodoStatus } from "../features/todo/todoSlice";
import { v4 as uuidv4 } from "uuid";
import "./Addform.scss";

import DateFnsUtils from "@date-io/date-fns";

import {
  MuiPickersUtilsProvider,
  KeyboardTimePicker,
  KeyboardDatePicker,
} from "@material-ui/pickers";

export const Addform = () => {
  const [title, settitle] = useState("");
  const [day, setDay] = useState("");

  const dispatch = useAppDispatch();
  const todolist = useAppSelector;

  const [selectedDate, setSelectedDate] = useState<Date | null>(
    new Date("2021-07-27T21:11:54")
  );

  //date choose handle
  const handleDateChange = (date: Date | null) => {
    setSelectedDate(date);
  };

  //submit add task
  const onSubmit = (e: any) => {
    e.preventDefault();

    if (!title) {
      alert("Please add a task");
      return;
    }

    const tdata: Todo = {
      id: uuidv4(),
      title: title,
      complated: false,
      endDate: selectedDate,
      createDate: new Date(),
    };

    dispatch(addTodo(tdata));

    settitle("");
    setDay("");
  };

  return (
    <div className="form-contain">
      <h1 className="text-blue-700 font-bold text-lg uppercase">add task</h1>
      <form className="add-form" onSubmit={onSubmit}>
        <div className="form-control">
          <label>Task</label>
          <input
            type="text"
            placeholder="Add Task"
            value={title}
            onChange={(e) => settitle(e.target.value)}
          />
        </div>
        <div className="form-control">
          <label>Day & Time</label>

          <MuiPickersUtilsProvider utils={DateFnsUtils}>
            <KeyboardDatePicker
              disableToolbar
              variant="inline"
              format="MM/dd/yyyy"
              margin="normal"
              id="date-picker-inline"
              label="Date picker inline"
              value={selectedDate}
              onChange={handleDateChange}
              KeyboardButtonProps={{
                "aria-label": "change date",
              }}
            />
          </MuiPickersUtilsProvider>
        </div>

        <input
          id="button"
          type="submit"
          value="Save Task"
          className="bg-purple-600 hover:bg-purple-700 focus:outline-none"
        />
      </form>
    </div>
  );
};
