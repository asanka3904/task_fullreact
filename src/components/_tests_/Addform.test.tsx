import React from "react";

import renderer from "react-test-renderer";
import { Addform } from "../Addform";
import { Provider } from "react-redux";
import { store } from "../../app/store";
import ReactDOM from "react-dom";

//Snapshot test
test("Addtodo", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Addform />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

//this is the dom test

describe("test render element", () => {
  let container: HTMLDivElement;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
    ReactDOM.render(
      <Provider store={store}>
        <Addform />
      </Provider>,
      container
    );
  });

  afterEach(() => {
    document.body.removeChild(container);
    container.remove();
  });

  it("test input feild", () => {
    const input = container.querySelectorAll("input");
    expect(input).toHaveLength(3);
  });
});
