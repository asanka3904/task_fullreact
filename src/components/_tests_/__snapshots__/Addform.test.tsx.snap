// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Addtodo 1`] = `
<div
  className="form-contain"
>
  <h1
    className="text-blue-700 font-bold text-lg uppercase"
  >
    add task
  </h1>
  <form
    className="add-form"
    onSubmit={[Function]}
  >
    <div
      className="form-control"
    >
      <label>
        Task
      </label>
      <input
        onChange={[Function]}
        placeholder="Add Task"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-control"
    >
      <label>
        Day & Time
      </label>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink={true}
          htmlFor="date-picker-inline"
          id="date-picker-inline-label"
        >
          Date picker inline
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
            disabled={false}
            id="date-picker-inline"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value="07/27/2021"
          />
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
          >
            <button
              aria-label="change date"
              className="MuiButtonBase-root MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        
      </div>
    </div>
    <input
      className="bg-purple-600 hover:bg-purple-700 focus:outline-none"
      id="button"
      type="submit"
      value="Save Task"
    />
  </form>
</div>
`;
